tidy3d/components/device/monitor_data.py

```diff
@@ -39,13 +39,77 @@ class HeatMonitorData(AbstractMonitorData, ABC):
         description="Symmetry center of the original simulation in x, y, and z.",
     )
 
-    @cached_property
-    def symmetry_expanded_copy(self) -> HeatMonitorData:
+    @property
+    def symmetry_expanded_copy(self) -> DeviceMonitorData:
         """Return copy of self with symmetry applied."""
         return self.copy()
 
+    def _symmetry_expanded_copy(self, property):
+        """Return the property with symmetry applied."""
+
+        # no symmetry
+        if all(sym == 0 for sym in self.symmetry):
+            return property
+
+        new_property = property
+
+        mnt_bounds = np.array(self.monitor.bounds)
+
+        if isinstance(new_property, SpatialDataArray):
+            data_bounds = [
+                [np.min(new_property.x), np.min(new_property.y), np.min(new_property.z)],
+                [np.max(new_property.x), np.max(new_property.y), np.max(new_property.z)],
+            ]
+        else:
+            data_bounds = new_property.bounds
+
+        dims_need_clipping_left = []
+        dims_need_clipping_right = []
+        for dim in range(3):
```

Comment:
this method is getting pretty complex and difficult to follow. Is there a way to simplifiy it?